lib/reply.js

```diff
@@ -129,19 +129,27 @@ function onSendEnd (reply, payload) {
     return
   }
 
-  if (!contentType && typeof payload === 'string') {
+  var isPayloadString = typeof payload === 'string'
+
+  if (!contentType && isPayloadString) {
     reply.res.setHeader('Content-Type', 'text/plain')
   } else if (reply._serializer) {
     payload = reply._serializer(payload)
+    if (typeof payload !== 'string' && !Buffer.isBuffer(payload)) {
```

Comment:
You can use `isPayloadString` also here.

Reply:
That's what I thought initially too, but the `payload` changes on the line above so it's a different value.

Reply:
Oops! you are right :P